tests/test_s3.py

```diff
@@ -5,14 +5,31 @@
 from botocore.exceptions import ClientError
 from moto import mock_s3
 from prefect import flow
+from pytest_lazyfixture import lazy_fixture
 
+from prefect_aws.client_parameters import AwsClientParameters
 from prefect_aws.s3 import s3_download, s3_list_objects, s3_upload
 
+aws_clients = [
+    lazy_fixture("aws_client_parameters_custom_endpoint"),
+    lazy_fixture("aws_client_parameters_empty"),
+]
+
 
 @pytest.fixture
-def s3_mock():
+def s3_mock(monkeypatch, request):
+    client_parameters = request.param
+    if client_parameters.endpoint_url:
+        monkeypatch.setenv("MOTO_S3_CUSTOM_ENDPOINTS", client_parameters.endpoint_url)
```

Comment:
Can you move the `client_parameters` setup to a different fixture?

Reply:
Added the client_paramters fixture